src/conn/conn_dhandle.c

```diff
@@ -134,7 +134,17 @@ __conn_dhandle_mark_dead(WT_SESSION_IMPL *session)
 	 * eviction and drain any blocks already queued.
 	 */
 	WT_RET(__wt_evict_file_exclusive_on(session, &evict_reset));
+
+	/* The handle is dead. */
 	F_SET(session->dhandle, WT_DHANDLE_DEAD);
+
+	/*
+	 * Reset the tree's eviction priority, and the tree is evictable by
+	 * definition.
+	 */
+	__wt_evict_priority_clear(session);
+	F_CLR(S2BT(session), WT_BTREE_NO_EVICTION);
```

Comment:
Resetting the no eviction flag is only necessary when ``evict_reset`` is returned from exclusive on. Otherwise we'll start considering pages in eviction when other code assumes the eviction server won't interfere.